internal/util/rand.go

```diff
@@ -2,21 +2,34 @@ package util
 
 import (
 	"math/rand"
+	"time"
 )
 
-func UniqRands(l int, n int) []int {
-	set := make(map[int]struct{})
-	nums := make([]int, 0, l)
-	for {
-		num := rand.Intn(n)
-		if _, ok := set[num]; !ok {
-			set[num] = struct{}{}
-			nums = append(nums, num)
-		}
-		if len(nums) == l {
-			goto exit
-		}
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+func UniqRands(quantity int, maxval int) []int {
+	if maxval < quantity {
+		quantity = maxval
 	}
-exit:
-	return nums
+
+	intSlice := make([]int, maxval)
+	for i := 0; i < maxval; i++ {
+		intSlice[i] = i
+	}
+
+	idx := 0
+	for i := 0; i < quantity; i++ {
+		idx = rand.Int()%maxval + i
+		// swap
+		swap(intSlice, i, idx)
```

Comment:
I don't think it's worth both a comment and a function for this.
I would just put the line of code that does the swap here.